jobsdb/jobsdb.go

```diff
@@ -615,28 +632,40 @@ becomes full OR in between during migration. DataSets are assigned numbers
 monotonically when added  to end. So, with just add to end, numbers would be
 like 1,2,3,4, and so on. Theese are called level0 datasets. And the Index is
 called level0 Index
-During migration, we add datasets in between. In the example above, if we migrate
+During internal migration, we add datasets in between. In the example above, if we migrate
 1 & 2, we would need to create a new DS between 2 & 3. This is assigned the
 the number 2_1. This is called a level1 dataset and the Index (2_1) is called level1
 Index. We may migrate 2_1 into 2_2 and so on so there may be multiple level 1 datasets.
 
 Immediately after creating a level_1 dataset (2_1 above), everything prior to it is
 deleted. Hence there should NEVER be any requirement for having more than two levels.
+
+There is an exception to this. In case of cross node migration during a scale up/down,
+we continue to accept new events in level0 datasets. To maintain the ordering guarantee,
+we write the imported jobs to the previous level1 datasets. Now if an internal migration
+is to happen on one of the level1 dataset, we ahve to migrate them to level2 dataset
+
+Eg. When the node has 1, 2, 3, 4 data sets and an import is triggered, new events start
+going to 5, 6, 7... so on. And the imported data start going to 4_1, 4_2, 4_3... so on
+Now if an internal migration is to happen and we migrate 1, 2, 3, 4, 4_1, we need to
+create a newDS between 4_1 and 4_2. This is assigned to 4_1_1, 4_1_2 and so on.
 */
 
 func (jd *HandleT) mapDSToLevel(ds dataSetT) (int, []int) {
 	indexStr := strings.Split(ds.Index, "_")
-	if len(indexStr) == 1 {
-		indexLevel0, err := strconv.Atoi(indexStr[0])
+	//Currently we don't have a scenario where we need more than 3 levels.
```

Comment:
@soumyadebm @SumanthPuram @chandumlg This change is regarding ds numbering. Please review this.